lib/classes/recordutils/image.php

```diff
@@ -243,9 +243,9 @@ public static function stamp(Application $app, \media_subdef $subdef)
             $texts = $xpprefs->query('text', $stamp);
             $fontsize = "100%";
             for ($i = 0; $i < $texts->length; $i++) {
-                if (($tmpfontsize = trim($texts->item($i)->getAttribute('size'))) != '') {
+                if (((int)$tmpfontsize = trim($texts->item($i)->getAttribute('size'))) != '') {
```

Comment:
Same bad original code. Better to retro-analyse (beware of "%"), and rewrite with 2 variables, one for str, one for numeric.
Do not make complex multiple casts, use 4000.0 to enforce float expression.